dlrover/trainer/torch/flash_checkpoint/full_ckpt_engine.py

```diff
@@ -112,6 +112,10 @@ def save_to_memory(self, step, state_dict, paths: Dict[str, str]):
                 ["model_states", "optim_states"] of the state dict and
                 the value is the path of storage to save.
         """
+        if self._checkpoint_event_step > 0:
+            notify_event = self._notify_queue.get()
+            assert notify_event.step == self._checkpoint_event_step
```

Comment:
what's ur intention if assert is false? 
In my personal opinion, this situation should be handled softly, such as skipping this step and providing a log explanation.